### PR TITLE
Include filename in Content-Disposition header for Document downloads

### PR DESCRIPTION
**Pull request by brownbearsoftware**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt [hunk 1]
```diff
@@ -38,14 +39,16 @@ class DocumentsController(
...
-    }
+    val documentMetaData = documentsMetaData.findDocument(documentId)
+      ?: throw NotFoundProblem(documentId, "Document")
 
     return ResponseEntity(
       StreamingResponseBody { outputStream ->
         offenderService.getDocument(crn, documentId, outputStream)
       },
+      HttpHeaders().apply {
+        put("Content-Disposition", listOf("attachment; filename=\"${documentMetaData.documentName}\""))
```
Comment (pezholio) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt:50]: Why is this a `listOf`?

**Review by pezholio — APPROVED** [review]
LGTM  - ta for the quick turnaround! 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt (hunk 1 above)
Comment (brownbearsoftware) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt:50]: Any HTTP header that can be represented as a comma seperated list can also be sent as multiple key-value pairs, e.g.

```
Cache-Control: no-cache, no-store
```

is the same as:

```
Cache-Control: no-cache
Cache-Control: no-store
```

So Spring types HttpHeaders as a Multi-Value map, i.e. `Map<String, List<String>>` to support the second case

**Comment by brownbearsoftware** [comment]
[Fetch documents from Delius](https://trello.com/c/2EtClS4K/897-fetch-documents-from-delius)

**Pull request merged**